specification/search/resource-manager/Microsoft.Search/stable/2020-03-13/search.json

```diff
@@ -1641,11 +1631,15 @@
           "externalDocs": {
             "url": "https://azure.microsoft.com/documentation/articles/search-sku-tier/"
           }
+        },
+        "identity": {
+          "$ref": "#/definitions/Identity",
+          "description": "The identity of the resource."
         }
```

Comment:
I get this error since the "location" and "tags" are not defined inherited correctly from ProxyResource

{
  code: 'REQUEST_VALIDATION_ERROR',
  id: 'OAV109',
  message: 'Found errors in validating the request for x-ms-example "SearchCreateOrUpdateService" in operation "Services_CreateOrUpdate".:\n' +
    ' Found errors in validating the request for x-ms-example "SearchCreateOrUpdateService" in operation "Services_CreateOrUpdate".',
  innerErrors: [
    {
      code: 'INVALID_REQUEST_PARAMETER',
      errors: [
        {
          code: 'OBJECT_ADDITIONAL_PROPERTIES',
          params: [ 'location' ],
          message: 'Additional properties not allowed: location',
          path: [],
          title: '{"path":["definitions","SearchService"],"position":{"line":1621,"column":22},"url":"Microsoft.Search\\\\stable\\\\2020-03-13\\\\search.json","directives":{}}',
          description: 'Describes an Azure Cognitive Search service and its current state.'
        },
        {
          code: 'OBJECT_ADDITIONAL_PROPERTIES',
          params: [ 'tags' ],
          message: 'Additional properties not allowed: tags',
          path: [],
          title: '{"path":["definitions","SearchService"],"position":{"line":1621,"column":22},"url":"Microsoft.Search\\\\stable\\\\2020-03-13\\\\search.json","directives":{}}',
          description: 'Describes an Azure Cognitive Search service and its current state.'
        }
      ],
      in: 'body',
      message: 'Invalid parameter (service): Value failed JSON Schema validation',
      name: 'service',
      path: [ 'parameters', 'service' ],
      schemaPath: '#/paths/~1subscriptions~1{subscriptionId}~1resourceGroups~1{resourceGroupName}~1providers~1Microsoft.Search~1searchServices~1{searchServiceName}/put/parameters/2',
      title: '{"path":["parameters","service"],"position":{"line":7,"column":16},"url":"Microsoft.Search/stable/2020-03-13/examples/SearchCreateOrUpdateService.json"}'
    }
  ],
  level: '\u001b[31merror\u001b[39m'
}
{
  '0': {
    code: 'RESPONSE_VALIDATION_ERROR',
    id: 'OAV108',
    message: 'Found errors in validating the response with statusCode "200" for x-ms-example "SearchCreateOrUpdateService" in operation "Services_CreateOrUpdate".',
    innerErrors: [
      {
        code: 'INVALID_RESPONSE_BODY',
        errors: [
          {
            code: 'OBJECT_ADDITIONAL_PROPERTIES',
            params: [ 'location' ],
            message: 'Additional properties not allowed: location',
            path: [],
            title: '#/definitions/SearchService',
            description: 'Describes an Azure Cognitive Search service and its current state.',
            position: { line: 1621, column: 22 },
            url: 'Microsoft.Search\\stable\\2020-03-13\\search.json',
            directives: {},
            jsonPosition: { line: 24, column: 15 },
            jsonUrl: 'Microsoft.Search/stable/2020-03-13/examples/SearchCreateOrUpdateService.json'
          },
          {
            code: 'OBJECT_ADDITIONAL_PROPERTIES',
            params: [ 'tags' ],
            message: 'Additional properties not allowed: tags',
            path: [],
            title: '#/definitions/SearchService',
            description: 'Describes an Azure Cognitive Search service and its current state.',
            position: { line: 1621, column: 22 },
            url: 'Microsoft.Search\\stable\\2020-03-13\\search.json',
            directives: {},
            jsonPosition: { line: 24, column: 15 },
            jsonUrl: 'Microsoft.Search/stable/2020-03-13/examples/SearchCreateOrUpdateService.json'
          }
        ],
        message: 'Invalid body: Value failed JSON Schema validation',
        path: []
      }
    ]
  },
  level: '\u001b[31merror\u001b[39m',
  message: 'Found errors in validating the response with statusCode "200" for x-ms-example "SearchCreateOrUpdateService" in operation "Services_CreateOrUpdate".:\n'
}

Reply:
Though, it will work if I inherit from Resource instead and add "location" and "tags" in the same fashion as I did with "Identity"

Just looking for a way to get "location" and "tags" from ProxyResource, or an explanation of why it does not work.

Reply:
Probably want to use `TrackedResource` here. This will result in breaking change by requiring `location` property. Please verify that is okay with ARM Reviewers. I would imagine that changing this to required is okay since requests would fail without it (ARM can't redirect to the proper region for the resource without it).